{{cookiecutter.project_slug}}/_/ci-services/definitions/.gitlab-ci.yml

```diff
@@ -41,20 +41,32 @@ stages:
                  -f "deployment/application/Dockerfile" .
   - docker push "${IMAGE}"
 
+.generate-secrets:
+  before_script:
+  - kubectl -n ${TARGET} get secret ${APPLICATION_NAME} ||
+    kubectl -n ${TARGET} create secret generic ${APPLICATION_NAME} --from-literal=DJANGO_SECRET_KEY=$(< /dev/urandom tr -dc A-Za-z0-9 | head -c50) |
```

Comment:
You need to specify `--dry-run -o yaml` to actually generate the YAML to stdout.

Reply:
I removed the apply step since we only create the secret if it does not exist.